environments/fetcher/fetcher.go

```diff
@@ -196,16 +200,50 @@ func (fetcher *Fetcher) Handler(w http.ResponseWriter, r *http.Request) {
 
 	}
 
-	// move tmp file to requested filename
-	err = os.Rename(tmpPath, filepath.Join(fetcher.sharedVolumePath, req.Filename))
+	targetFilename := filepath.Join(fetcher.sharedVolumePath, req.Filename)
+	// check file type here, if the file is a zip file unarchive it.
+	if archiver.Zip.Match(tmpPath) {
+		if req.FetchType == FETCH_SOURCE {
+			// builder accepts multiple builds at the same time, use `pkg.Name`-`pkg.ResourceVersion`
+			// to separate source pacakages for different functions.
+			targetFilename = filepath.Join(fetcher.sharedVolumePath, srcPkgFilename)
```

Comment:
targetFilename should always be req.Filename -- that way the caller controls the path of the download